sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/util/TelemetryHelper.java

```diff
@@ -0,0 +1,304 @@
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Classes;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Cpu;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.GarbageCollector;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.MemoryPools;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Threads;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Telemetry helper is used to monitor test execution and record stats.
+ */
+public class TelemetryHelper {
+    private final Tracer tracer;
+    private final ClientLogger logger;
+    private static final OpenTelemetry OTEL;
+    private static final AttributeKey<String> SCENARIO_NAME_ATTRIBUTE = AttributeKey.stringKey("scenario_name");
+    private static final AttributeKey<String> ERROR_TYPE_ATTRIBUTE = AttributeKey.stringKey("error.type");
+    private static final AttributeKey<Boolean> SAMPLE_IN_ATTRIBUTE = AttributeKey.booleanKey("sample.in");
+    private final String scenarioName;
+    private final Meter meter;
+    private final LongCounter closedPartitionCounter;
+    private final LongCounter initializedPartitionCounter;
+    private final LongCounter errorCounter;
+
+    private final DoubleHistogram runDuration;
+    private final Attributes commonAttributes;
+    private final Attributes canceledAttributes;
+
+    static {
+        // enables micrometer metrics from Reactor schedulers allowing to monitor thread pool usage and starvation
+        Schedulers.enableMetrics();
+        OTEL = init();
+    }
+
+    /**
+     * Creates an instance of telemetry helper.
+     * @param scenarioClass the scenario class
+     */
+    public TelemetryHelper(Class<?> scenarioClass) {
+        this.scenarioName = scenarioClass.getName();
+        this.tracer = OTEL.getTracer(scenarioName);
+        this.meter = OTEL.getMeter(scenarioName);
+        this.logger = new ClientLogger(scenarioName);
+        this.runDuration = meter.histogramBuilder("test.run.duration")
+            .setUnit("s")
+            .build();
+        this.commonAttributes = Attributes.of(SCENARIO_NAME_ATTRIBUTE, scenarioName);
+        this.canceledAttributes = Attributes.of(SCENARIO_NAME_ATTRIBUTE, scenarioName, ERROR_TYPE_ATTRIBUTE, "cancelled");
+        this.closedPartitionCounter = meter.counterBuilder("partition_closed").build();
+        this.initializedPartitionCounter = meter.counterBuilder("partition_initialized").build();
```

Comment:
for my own learning, is it a common pattern to use singular for counter name? for example, `partition_closed` instead of `partitions_closed`

Reply:
this name does not follow otel naming best practices in many ways :) I want to add those metrics to EH SDK and then will fix names. 

Here're naming best practices: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#pluralization